packages/openapi-generator/src/parser/operation.ts

```diff
@@ -30,9 +30,10 @@ export function parseOperation(
     refs
   );
 
-  const [pathParams, queryParams] = partition(
+  const [pathParams, headerParams, queryParams] = partitionThree(
     relevantParameters,
-    parameter => parameter.in === 'path'
+    parameter => parameter.in === 'path',
+    parameter => parameter.in === 'header'
   );
```

Comment:
[pp] I am not a fan of this implementation. I think I created the old one, but I think this would be easier to read:

```suggestion
const pathParams = relevantParameters.filter(parameter => parameter.in === 'path');
const headerParams = relevantParameters.filter(parameter => parameter.in === 'header');
const queryParams = relevantParameters.filter(parameter => parameter.in === 'query');
```